wc.c

```diff
@@ -23,34 +26,98 @@ void print_help() {
 }
 
 
-int main(int argc, char *argv[]) {
+struct Configuration {
+    bool char_counter;    // character counter
+    bool line_counter;    // line counter
+    bool word_counter;    // word counter
+
+    char* separator;      // define own separator for counting
+    char* filename;       // the file that should be parsed
+} config = {false, false, false, NULL};
```

Comment:
Missing `NULL` in init
```suggestion
} config = {false, false, false, NULL, NULL};
```